tests/periph_pwm/main.c

```diff
@@ -20,42 +20,135 @@
  * every 1s on every channel.
  *
  * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Semjon Kerner <semjon.kerner@fu-berlin.de>
  *
  * @}
  */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <limits.h>
 
 #include "xtimer.h"
+#include "shell.h"
 #include "timex.h"
 #include "periph/pwm.h"
 
-#define INTERVAL    (10LU * US_PER_MS) /* 10 ms */
-#define STEP        (10)
+#define OSC_INTERVAL    (10LU * US_PER_MS) /* 10 ms */
+#define OSC_STEP        (10)
+#define OSC_MODE        PWM_LEFT
+#define OSC_FREQU       (1000U)
+#define OSC_STEPS       (1000U)
+#define PWR_SLEEP       (1U)
 
-#define MODE        PWM_LEFT
-#define FREQU       (1000U)
-#define STEPS       (1000U)
+static uint32_t initiated;
 
-
-int main(void)
+static unsigned _get_dev(const char *dev_str)
 {
+    unsigned dev = (unsigned)atoi(dev_str);
+    if (dev >= PWM_NUMOF) {
+        printf("[pwm] Error: device PWM_DEV(%u) is unknown\n", dev);
+        return UINT_MAX;
+    }
+    return dev;
+}
+
+int init(int argc, char** argv) {
```

Comment:
here and below: make all shell command functions `static`. Also it is a good idea to prefex their names with an underscore.